src/passes/pass.cpp

```diff
@@ -483,6 +483,7 @@ void PassRunner::addDefaultGlobalOptimizationPostPasses() {
   } else {
     add("simplify-globals");
   }
+  add("vacuum"); // simplify-globals can generate extra nops
```

Comment:
If we are optimizing at `-O2` or above then we would be running `simplify-globals-optimizing` anyhow. We should make sure that pass will in fact optimize if we nop'ed an unread global, and then we don't need to add this vacuum (which isn't enough anyhow - the vacuum might open up more possibilities to optimize, and only the `-optimize` variant can handle).

Reply:
Are you saying that `simplify-globals-optimizing` should have take care of vacuming the nops?  Because it didn't seem too.  If it remember correctly, without this vacume the `-O2` tests also contained extra nops.

Reply:
Yeah, we may need to fix it so it does. The optimizing variant checks if we did "work" in the pass, then optimizes if so. Maybe we haven't told it about the nop changes?